src/lib/data/index.ts

```diff
@@ -268,3 +268,134 @@ export function useSettings(): {
 
     return { settings, updateSettings, loadSettings, handlers };
 }
+
+export interface AlertList {
+    dateKeys: string[];
+    [date: string]: string[];
+}
+
+export const defaultAlertList = {
+    dateKeys: [],
+};
+
+export function useAlertList(): {
+    alertList: AlertList;
+    addAlert: (date: string, id: string) => void;
+    removeAlert: (date: string, id: string) => void;
+    removeDateKey: (date: string) => void;
+    loadAlertList: () => Promise<void>;
+} {
+    const storage = useStorage();
+    const [alertList, setAlertList] = useState<AlertList>(defaultAlertList);
+
+    const removeDateKey = useCallback(
+        (date: string) => {
+            setAlertList((state) => {
+                const dateKeys = new Set<string>(state.dateKeys);
+                const cleanState: AlertList = { ...defaultAlertList };
+
+                dateKeys.delete(date);
+
+                dateKeys.forEach((dk: string) => {
+                    // We're picking all the dateKeys not including the current one
+                    if (state[dk] && dk !== date) {
```

Comment:
Won't this always be true if `dataKeys` is just a set from the state's `dataKeys` (so `state[dk] should always be defined) and you're deleting the entry for the passed-in `date` (so `dk` would never equal `date`)?